### PR TITLE
LPD-3348 Allow saving expiredDates in the past

### PR DESCRIPTION
**Pull request by dsitu**
https://liferay.atlassian.net/browse/LPD-3348

### CODE REVIEW

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -2507,24 +2507,26 @@ public CPDefinition updateStatus(
...
-				cpDefinition.setExpirationDate(null);
-			}
+			cpDefinition.setStatus(WorkflowConstants.STATUS_EXPIRED);
+
+			status = WorkflowConstants.STATUS_EXPIRED;
 		}
 
-		if (status == WorkflowConstants.STATUS_EXPIRED) {
+		if ((status == WorkflowConstants.STATUS_EXPIRED) &&
+			((expirationDate == null) || expirationDate.after(date))) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:2521]: this way is not possible to save the date in the past, is it?

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPDefinitionLocalServiceTest.java [hunk 2]
```diff
@@ -357,6 +362,50 @@ public void testAddCPDefinitionWithIgnoreSKUCombinationsAndDefaultInstance()
 		Assert.assertEquals(1, approvedCPInstances);
 	}
 
+	@Test
+	public void testAddExpiredCPDefinition() throws Exception {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPDefinitionLocalServiceTest.java:366]: please add a test also for the updateStatus method, that was changed too..

**Comment by dsitu** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPDefinitionLocalServiceTest.java (hunk 2 above)
Comment (dsitu) [reply modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPDefinitionLocalServiceTest.java:366]: Added 3 more test that should cover the different cases.

**Comment by dsitu** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you for adding tests, that helped understanding these different scenarios!

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**